### ISSUE TITLE
Jounald always enable audit even in third state

### ISSUE THREAD

**Issue by seblu** [systemd/systemd#19822]
**systemd version the issue has been seen with**
 > systemd 248 (248.3-seblu)
+PAM +AUDIT -SELINUX -APPARMOR -IMA +SMACK +SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN +IPTC +KMOD +LIBCRYPTSETUP +LIBFDISK +PCRE2 +PWQUALITY +P11KIT +QRENCODE +BZIP2 +LZ4 +XZ +ZLIB +ZSTD +XKBCOMMON +UTMP -SYSVINIT default-hierarchy=unified

**Used distribution**
 > Arch Linux

**Linux kernel version used** (`uname -a`)

 > `Linux hostname 5.12.9-arch1-1 #1 SMP PREEMPT Thu, 03 Jun 2021 11:36:13 +0000 x86_64 GNU/Linux`

**CPU architecture issue was seen on**
 > x86_64

**Steps to reproduce the problem**
-  Disable kernel audit. (e.g `auditctl -e 0 `)
 - Put Audit=-1 (or Audit=) in `/etc/systemd/journald.conf`
 - Run `systemd-journald`

**Unexpected behaviour you saw**
```
# SYSTEMD_LOG_LEVEL=debug /usr/lib/systemd/systemd-journald
/etc/systemd/journald.conf:18: Failed to parse boolean value, ignoring: -1
Auditing in kernel turned on.
```

**Expected behavior you didn't see**
As documented in [jounrald.conf.xml#L424](https://github.com/systemd/systemd/blob/efe76b273a40d96ecd51f88ab22aa25fb8e8d358/man/journald.conf.xml#L424), an "unset" value should prevent journald to enable/disable the kernel audit system.

**Notes**
- The tristate behavior for audit was added in 511e03a3eedb7613beb0ba59f98fdc1dd753aced
- The comment in `config_parse_tristate` claims the special value is -1. I guess it's the value referred as unset in the `jounald.conf` manpage. Anyway this value has neither special handling, neither be returned by the function.
- The function `config_parse_tristate` always call `parse_boolean` with the provided rvalue. When not a boolean `parse_boolean` always return `-EINVAL`. As a consequence,  `config_parse_tristate` always display a message warning message and let the value to its default. In the `Audit` case, the default value is enabled, so the tristate mean enabled and not to keep the kernel state.
- No warning message should be issued when Audit=-1 (or the correct tristate value).

**Comment by yuwata** [comment]
`config_parse_tristate()` only accepts an empty string or boolean string. `-1` is refused. 
```
If unset it will neither enable nor disable it, leaving the previous state unchanged.
```
Here `unset` means setting an empty string. That is, `Audit=`.

**Comment by seblu** [comment]
> Here `unset` means setting an empty string. That is, `Audit=`.

Ok, that was my first guess before reading the comment and is more coherent with other systemd behavior.

Remains that currently `config_parse_tristate()` doesn't handle **-1** and **""** differently.

**Comment by yuwata** [comment]
Ugh. I have not noticed that it neither accept an empty string... Fix is waiting in #19823.
